user/serializers.py

```diff
@@ -1,18 +1,242 @@
 from django.contrib.auth import authenticate
+from django.contrib.auth.password_validation import validate_password
+from django.db import transaction
+from django.utils.http import urlsafe_base64_decode
+from django.utils.translation import gettext
 from rest_framework import serializers
 
+from main.token import TokenManager
+from user.models import User
+from user.utils import (
+    resend_account_activation,
+    send_account_creation,
+    send_password_reset,
+)
+
 
 class LoginSerializer(serializers.Serializer):
     email = serializers.CharField()
     password = serializers.CharField(write_only=True)
 
     def validate(self, attrs):
-        # NOTE: authenticate only works for active users
         authenticate_user = authenticate(
             email=attrs["email"].lower(),
             password=attrs["password"],
         )
-        # User doesn't exists in the system.
         if authenticate_user is None:
             raise serializers.ValidationError("No active account found with the given credentials")
         return {"user": authenticate_user}
+
+
+def validate_token(attrs, token_generator) -> User:
+    try:
+        uid = urlsafe_base64_decode(attrs["uuid"]).decode("utf-8")
+        user = User.objects.get(pk=uid)
+    except (
+        TypeError,
+        ValueError,
+        OverflowError,
+        User.DoesNotExist,
+    ):
+        user = None
+
+    if user is not None and token_generator.check_token(user, attrs["token"]):
+        return user
+    raise serializers.ValidationError(gettext("Invalid or expired token"))
+
+
+class AddUserSerializer(serializers.Serializer):
+    emails = serializers.CharField(required=True)
+
+    def validate(self, validated_data):
+        emails = [email.strip().lower() for email in validated_data["emails"].split(",")]
+        registered_emails = User.objects.filter(email__in=emails).values_list("email", flat=True)
+        new_emails = list(set(emails) - set(registered_emails))
+        validated_data["new_emails"] = new_emails
+        return validated_data
+
+    def create(self, validated_data):
+        new_emails = validated_data.get("new_emails")
+        if not new_emails:
+            return None
+        new_users = []
+        with transaction.atomic():
+            for email in new_emails:
+                user = User.objects.create_user(
+                    email=email,
+                    password=None,
+                    is_active=False,
+                )
+                new_users.append(user)
+                transaction.on_commit(lambda user=user: send_account_creation(user))
```

Comment:
Pass User id instead of user object.
```suggestion
                transaction.on_commit(lambda: send_account_creation(user.id))
```